### DISCUSSION TITLE
Is it possible to simulate transaction with a transfer amount greater than account balance?

### DISCUSSION THREAD

**Discussion by pauld9x** [aptos-labs/aptos-developer-discussions#256] [Questions]
### Describe your question in detail.

Hi, I need to simulate a transaction with a large amount (larger than my APT balance) on the Aptos mainnet.
Thanks!

### What have you tried or looked at?  Or how can we reproduce the error?

With my account, I get the error "INSUFFICIENT BALANCE".
and,
I also tried using an account on explorer with a large amount but got the error "INVALID_AUTH_KEY"
`const account = new AptosAccount(undefined, customAddress)`

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xbe1** [comment]
I think this is possible with evm using [viem](https://viem.sh/), so definitely a nice feature to have on aptos

**Reply by pauld9x** [reply]
yes, I used it on EVM chain. And I'm looking for a way to use in Aptos

**Reply by gregnazario** [reply]
Currently, the simulation API works on actual existing state, without overrides to balance.  It's an interesting thing to have for development purposes, will note that.

As for being able to simulate any random account (the invalid auth key error you have), we're working on a way where you can simulate without giving a PublicKey that has to match the account.

Any other ideas for simulation please put them here!  Some of my thoughts are ability to simulate multiple chained calls and no gas limits

**Reply by 0xbe1** [reply]
> ability to simulate multiple chained calls

@gregnazario are you referring to "PTB"?
